fuel-tx/src/transaction/validity.rs

```diff
@@ -321,6 +322,24 @@ where
     Ok(())
 }
 
+pub(crate) fn check_owner<T>(tx: &T) -> Result<(), ValidityError>
+where
+    T: Chargeable,
+{
+    if let Some(owner) = tx.owner() {
+        let owner = usize::try_from(owner)
+            .map_err(|_| ValidityError::TransactionOwnerIndexDoesntExist)?;
+        if owner >= tx.inputs().len() {
+            Err(ValidityError::TransactionOwnerIndexDoesntExist)?
+        }
+        // SAFETY: `owner` is guaranteed to be a valid index because it was checked above.
+        if !&tx.inputs()[owner].input_owner().is_some() {
+            Err(ValidityError::TransactionOwnerInputHasNoOwner)?
+        }
```

Comment:
can we avoid manual safety check with something like:
```
        if tx
            .inputs()
            .get(owner)
            .and_then(|input| input.input_owner())
            .is_none()
        {
            return Err(ValidityError::TransactionOwnerInputHasNoOwner);
        }
```
?

Reply:
Ok I will change that. (not the whole code to mentionned because otherwise we will loose the two different errors)